Fix ChromeVox animated elements inaccessible

jeferson-sb opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/main/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Update elements with initial/ending opacity animation to `0.00001` to make sure ChromeVox (Chromebook's Screen Reader) 
can announce its content.

## Related Tickets & Documents
#12939 

## QA Instructions, Screenshots, Recordings

### UI accessibility concerns?

## Added tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

aitchiss commented on app/assets/stylesheets/articles.scss:
```diff
@@ -110,7 +110,7 @@
 @keyframes fade-in {
   0% {
-    opacity: 0;
+    opacity: 0.00001;
```
small change (blocking) - as we'll be trying to make sure we always use this value instead of `opacity: 0`, let's add a variable for it in `_variables.scss`, e.g. `$opacity-0: 0.0001`. Then we can use it in all the places implemented here

jeferson-sb replied:
fixed

aitchiss reviewed (changes requested):
Thank you for tackling this issue! Have you been able to test it OK with ChromeVox?

One other small change request - we also use the imported utility class `opacity-0` as well as explicitly setting the opacity directly in CSS. We should override this tailwind class so that we have

```css
.opacity-0 {
    opacity: 0.00001;
}
```

I think you could add this to `helpers.scss`

nickytonline reviewed (approved):
Thanks for the PR @jeferson-sb and making Forem more accessible! ♥️

ludwiczakpawel commented:
Could we use [CSS native variables](https://github.com/forem/forem/blob/main/app/assets/stylesheets/config/_variables.scss) instead of SCSS ones since we're slowly stepping away from these? 

> One other small change request - we also use the imported utility class opacity-0 as well as explicitly setting the opacity directly in CSS. We should override this tailwind class so that we have
> ...
> I think you could add this to helpers.scss

Just FYI looks like @nickytonline has already handled that in our [utility classes generator](https://github.com/forem/forem/blob/main/app/assets/stylesheets/config/_generator.scss#L37).

nickytonline commented:
> Could we use [CSS native variables](https://github.com/forem/forem/blob/main/app/assets/stylesheets/config/_variables.scss?rgh-link-date=2021-05-06T12%3A54%3A28Z) instead of SCSS ones since we're slowly stepping away from these?
> 
> > One other small change request - we also use the imported utility class opacity-0 as well as explicitly setting the opacity directly in CSS. We should override this tailwind class so that we have
> > ...
> > I think you could add this to helpers.scss
> 
> Just FYI looks like @nickytonline has already handled that in our [utility classes generator](https://github.com/forem/forem/blob/main/app/assets/stylesheets/config/_generator.scss?rgh-link-date=2021-05-06T12%3A54%3A28Z#L37).

Yeah, the `opacity-0` CSS utility class is separate from this though. We could have an `--opacity-0` CSS var that both consume. I didn't opt for a CSS variable in `_generator.scss` as we don't expose CSS vars in there at the moment, but we could.

aitchiss commented:
Hey @jeferson-sb - are you able to make the change that Pawel suggested above (to make this a CSS variable instead of a Sass one)? I think all you would need to do is remove the `$opacity-0` variable and instead create a CSS one in `_variables.scss` e.g. `--opacity-0` (and then change the references to it throughout the other files).

I think this will also fix the build errors we're seeing at the moment, which are due to the sass variables not being imported.

jeferson-sb commented:
Hi @aitchiss I've added `--opacity-0` as CSS variables and a few more values and removed the sass one. 
Thanks for the suggestions @ludwiczakpawel and @nickytonline 😃

aitchiss reviewed (approved):
Thanks for the changes - this looks great 🚀 

I'm re-running the failed build just now, as it looks like an unrelated flaky spec. All being well, we can get this merged 😄
